gradio/chat_interface.py

```diff
@@ -0,0 +1,269 @@
+            chatbot: an instance of the gr.Chatbot component to use for the chat interface. If not provided, a default gr.Chatbot component will be created.
+            textbox: an instance of the gr.Textbox component to use for the chat interface. If not provided, a default gr.Textbox component will be created.
+            examples: sample inputs for the function; if provided, appear below the chatbot and can be clicked to populate the chatbot input.
+            cache_examples: If True, caches examples in the server for fast runtime in examples. The default option in HuggingFace Spaces is True. The default option elsewhere is False.
+            title: a title for the interface; if provided, appears above chatbot in large font. Also used as the tab title when opened in a browser window.
+            description: a description for the interface; if provided, appears above the chatbot and beneath the title in regular font. Accepts Markdown and HTML content.
+            theme: Theme to use, loaded from gradio.themes.
+            css: custom css or path to custom css file to use with interface.
+            analytics_enabled: Whether to allow basic telemetry. If None, will use GRADIO_ANALYTICS_ENABLED environment variable if defined, or default to True.
+            submit_btn: Text to display on the submit button. If None, no button will be displayed. If a Button object, that button will be used.
+            retry_btn: Text to display on the retry button. If None, no button will be displayed. If a Button object, that button will be used.
+            delete_last_btn: Text to display on the delete last button. If None, no button will be displayed. If a Button object, that button will be used.
+            clear_btn: Text to display on the clear button. If None, no button will be displayed. If a Button object, that button will be used.
+        """
+        super().__init__(
+            analytics_enabled=analytics_enabled,
+            mode="chat_interface",
+            css=css,
+            title=title or "Gradio",
+            theme=theme,
+        )
+        self.fn = fn
+        self.examples = examples
+        self.cache_examples = cache_examples
+        self.buttons: list[Button] = []
+
+        with self:
+            if title:
+                Markdown(
+                    f"<h1 style='text-align: center; margin-bottom: 1rem'>{self.title}</h1>"
+                )
+            if description:
+                Markdown(description)
+
+            with Group():
+                if chatbot:
+                    self.chatbot = chatbot.render()
+                else:
+                    self.chatbot = Chatbot(label="Input")
+                if textbox:
+                    self.textbox = textbox.render() 
+                else: 
+                    self.textbox = Textbox(show_label=False, placeholder="Type a message...")
+            
+            with Row():
+                if submit_btn:
```

Comment:
I think the UI would look a lot sleeker if the submit button was next to the textbox

<img width="944" alt="Screenshot 2023-07-13 at 10 57 11 AM" src="https://github.com/gradio-app/gradio/assets/41651716/d666d81b-3c3e-486a-8f0c-885372448cd1">

Similar to ChatGPT, HuggingChat, and iOS

Reply:
Ok let me play around with that and see how it looks on different screen sizes